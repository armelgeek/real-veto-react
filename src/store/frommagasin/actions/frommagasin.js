import {
  ADD_QTY_FROM_MAGASIN_ITEM,
  ADD_FROM_MAGASIN,
  CLEAR_FROM_MAGASIN,
  MINUS_QTY_FROM_MAGASIN_ITEM,
  REMOVE_FROM_TO_MAGASIN,
  SET_FROM_MAGASIN_ITEMS,
  ADD_QTY_PORTION_FROM_MAGASIN_ITEM,
  MINUS_QTY_PORTION_FROM_MAGASIN_ITEM,
  ADD_QTY_NUMBER_FROM_MAGASIN_ITEM,
} from "../constants";

export const setFromMagasinItems = (items = []) => ({
  type: SET_FROM_MAGASIN_ITEMS,
  payload: items,
});

export const addFromMagasin = (product) => ({
  type: ADD_FROM_MAGASIN,
  payload: product,
});

export const removeFromMagasin = (id) => ({
  type: REMOVE_FROM_TO_MAGASIN,
  payload: id,
});

export const clearFromMagasin = () => ({
  type: CLEAR_FROM_MAGASIN,
});

export const addQtyFromMagasinItem = (id, qtt) => ({
  type: ADD_QTY_FROM_MAGASIN_ITEM,
  payload: { id: id, qtt: qtt },
});

export const setQtyFromMagasinItem = (id, value) => ({
  type: ADD_QTY_NUMBER_FROM_MAGASIN_ITEM,
  payload: {
    id: id,
    value: value,
  },
});
export const minusQtyFromMagasinItem = (id) => ({
  type: MINUS_QTY_FROM_MAGASIN_ITEM,
  payload: { id: id },
});

export const addQtyFromMagasinPortionItem = (id, value) => ({
  type: ADD_QTY_PORTION_FROM_MAGASIN_ITEM,
  payload: { id: id, value: value },
});

export const minusQtyFromMagasinPortionItem = (id) => ({
  type: MINUS_QTY_PORTION_FROM_MAGASIN_ITEM,
  payload: { id: id },
});
