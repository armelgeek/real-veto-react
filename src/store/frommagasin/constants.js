export const SET_FROM_MAGASIN_ITEMS = 'SET_FROM_MAGASIN_ITEMS';
export const ADD_FROM_MAGASIN = 'ADD_FROM_MAGASIN';
export const REMOVE_FROM_TO_MAGASIN = 'REMOVE_FROM_TO_MAGASIN';
export const CLEAR_FROM_MAGASIN = 'CLEAR_FROM_MAGASIN';
export const ADD_QTY_FROM_MAGASIN_ITEM = 'ADD_QTY_FROM_MAGASIN_ITEM';
export const MINUS_QTY_FROM_MAGASIN_ITEM = 'MINUS_QTY_FROM_MAGASIN_ITEM';
export const ADD_NUMBER_QTY_ITEM ='ADD_NUMBER_QTY_ITEM';
export const ADD_QTY_PORTION_FROM_MAGASIN_ITEM = "ADD_QTY_PORTION_FROM_MAGASIN_ITEM";
export const MINUS_QTY_PORTION_FROM_MAGASIN_ITEM = "MINUS_QTY_PORTION_FROM_MAGASIN_ITEM";
export const ADD_QTY_NUMBER_FROM_MAGASIN_ITEM = "ADD_QTY_NUMBER_FROM_MAGASIN_ITEM";
export const  MINUS_QTY_PORTION__DOSE_ITA_NEW_FROM_MAGASIN_ITEM = "MINUS_QTY_PORTION_DOSE_ITA_NEW_FROM_MAGASIN_ITEM";
export const ADD_QTY_PORTION_DOSE_UTA_NEW_FROM_MAGASIN_ITEM = "ADD_QTY_PORTION_DOSE_ITA_NEW_FROM_MAGASIN_ITEM";
export const ADD_QTY_PORTION_DOSE_SOVAX_FROM_MAGASIN_ITEM = "ADD_QTY_PORTION_DOSE_SOVAX_FROM_MAGASIN_ITEM";
export const MINUS_QTY_PORTION__DOSE_SOVAX_FROM_MAGASIN_ITEM="MINUS_QTY_PORTION__DOSE_SOVAX_FROM_MAGASIN_ITEM";
export const ADD_QTY_PORTION_PHYTO_COND_FROM_MAGASIN_ITEM="ADD_QTY_PORTION_PHYTO_COND_FROM_MAGASIN_ITEM";
export const MINUS_QTY_PORTION_PHYTO_COND_FROM_MAGASIN_ITEM="MINUS_QTY_PORTION_PHYTO_COND_FROM_MAGASIN_ITEM";

export const ADD_QTY_LITRE_NUMBER_FROM_MAGASIN_ITEM = "ADD_QTY_LITRE_NUMBER_FROM_MAGASIN_ITEM";
export const MINUS_QTY_LITRE_NUMBER_FROM_MAGASIN_ITEM = "MINUS_QTY_LITRE_NUMBER_FROM_MAGASIN_ITEM";