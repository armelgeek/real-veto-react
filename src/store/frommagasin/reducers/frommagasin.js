import {
  ADD_QTY_FROM_MAGASIN_ITEM,
  ADD_FROM_MAGASIN,
  CLEAR_FROM_MAGASIN,
  MINUS_QTY_FROM_MAGASIN_ITEM,
  REMOVE_FROM_TO_MAGASIN,
  SET_FROM_MAGASIN_ITEMS,
  MINUS_QTY_PORTION_FROM_MAGASIN_ITEM,
  ADD_QTY_PORTION_FROM_MAGASIN_ITEM,
  ADD_QTY_NUMBER_FROM_MAGASIN_ITEM,
  ADD_QTY_PORTION_DOSE_UTA_NEW_FROM_MAGASIN_ITEM,
  MINUS_QTY_PORTION__DOSE_ITA_NEW_FROM_MAGASIN_ITEM,
  MINUS_QTY_PORTION__DOSE_SOVAX_FROM_MAGASIN_ITEM,
  ADD_QTY_PORTION_DOSE_SOVAX_FROM_MAGASIN_ITEM,
  ADD_QTY_PORTION_PHYTO_COND_FROM_MAGASIN_ITEM,
  MINUS_QTY_PORTION_PHYTO_COND_FROM_MAGASIN_ITEM,
} from "../constants";
import {
  handleMinusProduct,
  handleSoldQuantityCC,
} from "../../functions/function";
// eslint-disable-next-line import/no-anonymous-default-export
export default (state = [], action) => {
  switch (action.type) {
    case SET_FROM_MAGASIN_ITEMS:
      return action.payload;
    case ADD_FROM_MAGASIN:
      return state.some((product) => product.id === action.payload.id)
        ? state
        : [action.payload, ...state];
    case REMOVE_FROM_TO_MAGASIN:
      return state.filter((product) => product.id !== action.payload);
    case CLEAR_FROM_MAGASIN:
      return [];
    case ADD_QTY_FROM_MAGASIN_ITEM:
      return state.map((product) => {
        if (product.id === action.payload.id) {
          product.quantityParProduct = Number(action.payload.qtt);
        }
        return product;
      });

    case ADD_QTY_PORTION_FROM_MAGASIN_ITEM:
      return state.map((product) => {
        if (product.id == action.payload.id) {
            product.qttByCC = Number(action.payload.value);
        }

        return product;
      });
      id;
    case ADD_QTY_NUMBER_FROM_MAGASIN_ITEM:
      return state.map((product) => {
        if (product.id === action.payload.id) {
          product.quantityParProduct = Number(action.payload.value);
        }
        return product;
      });
    case ADD_QTY_PORTION_DOSE_UTA_NEW_FROM_MAGASIN_ITEM:
      return state.map((product) => {
        if (product.id === action.payload.id) {
          product.qttyspecificmirror = Number(action.payload.value);
          product.qttByCC = action.payload.value * 1 / 2;
        }
        return product;
      });

    case ADD_QTY_PORTION_DOSE_SOVAX_FROM_MAGASIN_ITEM:
      return state.map((product) => {
        if (product.id === action.payload.id) {
          product.qttyspecificmirror = Number(action.payload.value);
          product.qttByCC = (action.payload.value * 1) * 2;
        }
        return product;
      });
    case ADD_QTY_PORTION_PHYTO_COND_FROM_MAGASIN_ITEM:
      return state.map((product) => {
        if (product.id == action.payload.id) {
          product.qttByCC = Number(action.payload.value);
        }
        return product;
      });
    case MINUS_QTY_PORTION_PHYTO_COND_FROM_MAGASIN_ITEM:
      return state.map((product) => {
        if (product.id === action.payload.id) {
          product.qttByCC -= action.payload.qttccpvente;
        }
        return product;
      });
    default:
      return state;
  }
};
