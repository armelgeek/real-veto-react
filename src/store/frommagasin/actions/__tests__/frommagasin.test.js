import * as selectActions from "../frommagasin";
import configureStore from "redux-mock-store";
const mockStore = configureStore();
const store = mockStore();
describe("testing actions from magasin", () => {
  beforeEach(() => {
    store.clearActions();
  });
  it.skip("should set quantity product from magasin to orders", () => {
    store.dispatch(selectActions.setQtyFromMagasinItem(1, 1));
    expect(store.getActions()).toMatchSnapshot();
  });
});
