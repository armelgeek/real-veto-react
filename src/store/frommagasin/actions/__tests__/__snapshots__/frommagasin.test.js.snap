// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing actions from magasin should set quantity product from magasin to orders 1`] = `
Array [
  Object {
    "payload": Object {
      "id": 1,
      "value": 1,
    },
    "type": "ADD_QTY_NUMBER_FROM_MAGASIN_ITEM",
  },
]
`;
